src/bin/hub_manager/hub_manager.cpp

```diff
@@ -271,3 +275,141 @@ void jaiabot::apps::HubManager::handle_dive_packet(const jaiabot::protobuf::Dive
+
+            // The initial fragment is going to have more data
+            if (command.plan().has_start() && fragment_index == 0)
+            {
+                command_fragment.mutable_plan()->set_start(command.plan().start());
+            }
+
+            if (command.plan().has_movement() && fragment_index == 0)
+            {
+                command_fragment.mutable_plan()->set_movement(command.plan().movement());
+            }
+
+            if (command.plan().has_recovery() && fragment_index == 0)
+            {
+                protobuf::MissionPlan_Recovery* recovery = new protobuf::MissionPlan_Recovery();
+                recovery->ParseFromString(command.plan().recovery().SerializeAsString());
+                command_fragment.mutable_plan()->set_allocated_recovery(recovery);
+            }
+
+            command_fragment.mutable_plan()->set_fragment_index(fragment_index);
+
+            command_fragment.mutable_plan()->set_expected_fragments(command_fragments_expected);
+
+            goal_max_index = goal_max_index + goal_max_size;
+
+            glog.is_debug1() && glog << group("main") << "Goal Index: " << goal_max_index
+                                     << ", max size: " << goal_max_size
+                                     << ", Total goal size: " << command.plan().goal_size()
+                                     << ", Goal index: " << goal_index << std::endl;
+
+            // Loop through goals and add to fragment
+            for (; goal_index < command.plan().goal_size(); goal_index++)
+            {
+                if (goal_index < goal_max_index)
+                {
+                    glog.is_debug1() && glog << group("main") << "Goal max size: " << goal_max_size
+                                             << ", goal index: " << goal_index
+                                             << ", Total goal size: " << command.plan().goal_size()
+                                             << std::endl;
+
+                    protobuf::MissionPlan::Goal* goal = command_fragment.mutable_plan()->add_goal();
+                    if (command.plan().goal(goal_index).has_name())
+                    {
+                        goal->set_name(command.plan().goal(goal_index).name());
+                    }
+                    if (command.plan().goal(goal_index).has_task())
+                    {
+                        protobuf::MissionTask* task = new protobuf::MissionTask();
+                        task->ParseFromString(
+                            command.plan().goal(goal_index).task().SerializeAsString());
+                        goal->set_allocated_task(task);
+                    }
+
+                    protobuf::GeographicCoordinate* coord = new protobuf::GeographicCoordinate();
+                    coord->ParseFromString(
+                        command.plan().goal(goal_index).location().SerializeAsString());
+                    goal->set_allocated_location(coord);
```

Comment:
See above, no need for SerialToString and then Parse. Very inefficient and confusing:

```
*goal->mutable_location() = command.plan.goal(goal_index).location();
```

Reply:
Ya, this is way better!